client/src/views/Settings/Tabs/EditPrivacy/index.vue

```diff
@@ -75,21 +75,197 @@
         </q-item-section>
       </q-item>
 
+      <q-separator spaced />
+      <q-item-label header>Black list</q-item-label>
+      <div class="row justify-between">
+        <div
+          id="input-search"
+          ref="search"
+          class="col"
+          >
+          <q-input v-model="nickname" @input="searchByNick" />
+          <block-search
+            v-if="showSearch"
+            id="search-result"
+            :results="nicknameSearchResults"
+            @handleChoose="handleChoose"
+          />
+        </div>
+        <q-btn
+          @click="blockUser"
+          id="block-button"
+          class="col"
+          :disabled='!enableBlock'
+        >Block</q-btn>
+      </div>
+      <p v-if="userToBlockNick">{{userToBlockNick}}  <q-btn
+        round
+        color = 'teal'
+        size = 'xs'
+        icon = 'close'
+        @click = "userToBlockNick = ''"
+      /></p>
+      <p id="error-message" v-if="errorMessage">{{errorMessage}}  <q-btn
+        round
+        color = 'red'
+        size = 'xs'
+        icon = 'close'
+        @click = "errorMessage = ''"
+      /></p>
+      <blocked-list id="blocked-list" @unblock="unblockUser" :list="blockedProfilesList"></blocked-list>
     </q-list>
   </div>
 </template>
 
 <script>
+import BlockSearch from '@/components/BlockSearch.vue'
+import BlockedList from '@/components/BlockedList.vue'
+import debounce from 'lodash/debounce'
+import { searchByNick, profileAction, getProfile } from '@/services/profile.js'
+
 export default {
   name: 'EditPrivacy',
+  components: {
+    BlockSearch,
+    BlockedList
+  },
   data () {
     return {
       comment: true,
       like: true,
       reply: true,
       uploadPost: true,
-      hideOffencive: false
+      hideOffencive: false,
+      nickname: '',
+      showSearch: true,
+      myid: this.$store.state.profile.myProfileId,
```

Comment:
How many times I need to repeat, use getters!

Reply:
Sorry, that was meant to be a simplified version before Open Night. Now it's with getter...